conan/tools/microsoft/visual.py

```diff
@@ -54,18 +54,21 @@ def generate(self, scope="build"):
 
 
 def vs_ide_version(conanfile):
+    """
+    Get the VS IDE version as string
+    """
     compiler = conanfile.settings.get_safe("compiler")
     compiler_version = (conanfile.settings.get_safe("compiler.base.version") or
                         conanfile.settings.get_safe("compiler.version"))
     if compiler == "msvc":
-        toolset_override = conanfile.conf["tools.microsoft.msbuild:vs_version"]
+        toolset_override = conanfile.conf.get("tools.microsoft.msbuild:vs_version")
         if toolset_override:
             visual_version = toolset_override
         else:
             visual_version = msvc_version_to_vs_ide_version(compiler_version)
     else:
         visual_version = compiler_version
-    return visual_version
+    return str(visual_version)
```

Comment:
This must be a string at this point, right? Why conversion?

Reply:
this `conanfile.conf.get("tools.microsoft.msbuild:vs_version")` returns a integer:

```
tools.microsoft.msbuild:vs_version = 17
```
Internally, it's managed as an integer. If you want to force it to be a string value, you'll have to write:

```
tools.microsoft.msbuild:vs_version = "17"
```
So I can change that line and apply the conversion to the original one:
```python
toolset_override = str(conanfile.conf.get("tools.microsoft.msbuild:vs_version", default=""))
```

Reply:
That is the thing, we can make this incorrect, and fail if it is not a string? All versions are always strings, not integer. Defining a version as an integer is not great, but that opens the door to define a version as a float like ``...:vs_version=17.3``, which looks correct on the surface, but managing such version as a float is really bad. So maybe we want to ban defining versions as integers too?